lib/functions/image/partitioning.sh

```diff
@@ -98,7 +98,7 @@ function prepare_partitions() {
 		local uefipart=$((next++))
 	fi
 	# Check if we need boot partition
-	if [[ $BOOTSIZE != "0" && ( -n $BOOTFS_TYPE || $ROOTFS_TYPE != ext4 || $CRYPTROOT_ENABLE == yes ) ]]; then
+	if [[ $BOOTSIZE != "0" && ( -n $BOOTFS_TYPE || $ROOTFS_TYPE != ext4 || $CRYPTROOT_ENABLE == yes || $LVM_ENABLE == yes ) ]]; then
```

Comment:
I think we don't need to add a new variable here. Just do something like BOOTFS_TYPE=$ROOTFS_TYPE or to ext4 in extension_prepare_config and it should be fine.

Reply:
@viraniac I am not entirely sure that would work. BOOTFS_TYPE=ROOTFS_TYPE would break cloud-init, because it requires fat, forcing ext4 would have a similar impact, I think.

I think this all depends on whether CRYPTROOT will be moved to an extension or not, then the extensions could set an specific variable to signal that a BOOT partition is mandatory.

Reply:
> @viraniac I am not entirely sure that would work. BOOTFS_TYPE=ROOTFS_TYPE would break cloud-init

I don't think so. Things can be easily handled if you want them to be handled. The following in your extension's extension_prepare_config should be able to easily handle the same.

```[[ -n BOOTFS_TYPE ]] || BOOTFS_TYPE=$ROOTFS_TYPE```

Reply:
> I think this all depends on whether CRYPTROOT will be moved to an extension or not

There is an extension for cryptroot, file is named fs-cryptroot-support.sh. Its just not all of the code was moved to it. Hence I did asked you before to move the cryptroot related code there as you are already working on this.

Reply:
@viraniac , I see, then I can move some more of it.